internal/pkg/cmd/authenticated_cli_command.go

```diff
@@ -198,15 +201,15 @@ func (c *AuthenticatedCLICommand) GetSchemaRegistryClient() (*schemaregistry.Cli
 				configuration.DefaultHeader = map[string]string{"target-sr-cluster": clusters[0].GetId()}
 				configuration.BasePath = clusters[0].Spec.GetHttpEndpoint()
 
-				dataplaneToken, err := auth.GetDataplaneToken(c.Context.GetState(), c.Context.GetPlatformServer())
+				dataplaneToken, err := auth.GetDataplaneToken(ctx.GetState(), ctx.GetPlatformServer())
 				if err != nil {
 					return nil, err
 				}
 
-				c.schemaRegistryClient = schemaregistry.NewClient(configuration, dataplaneToken)
+				c.schemaRegistryClient = schemaregistry.NewClientWithToken(configuration, dataplaneToken)
 			} else {
 				// Used by `asyncapi export`, `asyncapi import`, `kafka client-config create`, `kafka topic consume`, and `kafka topic produce`
-				schemaRegistryEndpoint, err := c.Flags().GetString("schema-registry-endpoint")
+				schemaRegistryEndpoint, err := cmd.Flags().GetString("schema-registry-endpoint")
```

Comment:
Why do we need to pass in `cmd` instead of using `c`?

Reply:
`c` is created in `kafka topic` which doesn't have `--schema-registry-endpoint`, but `cmd` refers to `kafka topic consume`, which does.